code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -354,262 +363,265 @@
-						if(!Adjacent(user) || !new_dna_hash)
-							return
-						dna_hash = new_dna_hash
-						to_chat(user, "<span class='notice'>DNA hash changed to [new_dna_hash].</span>")
-						RebuildHTML()
-
-					if("Fingerprint Hash")
-						var/default = "\[UNSET\]"
-						if(ishuman(user))
-							var/mob/living/carbon/human/H = user
-							if(H.dna)
-								default = md5(H.dna.uni_identity)
-
-						var/new_fingerprint_hash = tgui_input_text(user, "What fingerprint hash would you like to be written on this card?", "Agent Card Fingerprint Hash", default)
-						if(!Adjacent(user) || !new_fingerprint_hash)
-							return
-						fingerprint_hash = new_fingerprint_hash
-						to_chat(user, "<span class='notice'>Fingerprint hash changed to [new_fingerprint_hash].</span>")
-						RebuildHTML()
-
-					if("Reset Access")
-						var/response = tgui_alert(user, "Are you sure you want to reset access saved on the card?", "Reset Access", list("No", "Yes"))
-						if(response == "Yes")
-							access = initial_access.Copy() // Initial() doesn't work on lists
-							to_chat(user, "<span class='notice'>Card access reset.</span>")
-
-					if("Delete Card Information")
-						var/response = tgui_alert(user, "Are you sure you want to delete all information saved on the card?", "Delete Card Information", list("No", "Yes"))
-						if(response == "Yes")
-							name = initial(name)
-							registered_name = initial(registered_name)
-							icon_state = initial(icon_state)
-							sex = initial(sex)
-							age = initial(age)
-							assignment = initial(assignment)
-							associated_account_number = initial(associated_account_number)
-							blood_type = initial(blood_type)
-							dna_hash = initial(dna_hash)
-							fingerprint_hash = initial(fingerprint_hash)
-							photo = null
-							registered_user = null
-							to_chat(user, "<span class='notice'>All information has been deleted from \the [src].</span>")
-							RebuildHTML()
+/obj/item/card/id/syndicate/ui_data(mob/user)
+	var/list/data = list()
+	data["registered_name"] = registered_name
+	data["sex"] = sex
+	data["age"] = age
+	data["assignment"] = assignment
+	data["job_icon"] = ckey(get_job_name())
+	data["associated_account_number"] = associated_account_number
+	data["blood_type"] = blood_type
+	data["dna_hash"] = dna_hash
+	data["fingerprint_hash"] = fingerprint_hash
+	data["photo"] = photo
+	data["ai_tracking"] = untrackable
+	return data
+
+/obj/item/card/id/syndicate/ui_static_data(mob/user)
+	var/list/data = list()
+	var/list/idcard_skins = list()
+	for(var/idcard_skin in get_all_card_skins())
+		idcard_skins.Add(list(list(
+			"name" = idcard_skin,
+		)))
+	data["idcards"] = idcard_skins
+	return data
+
+
+/obj/item/card/id/syndicate/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/idcards),
+		get_asset_datum(/datum/asset/spritesheet/orbit_job),
+	)
+
+/obj/item/card/id/syndicate/ui_state(mob/user)
+	return GLOB.default_state
```

Comment:
Don't need to redeclare this, right? Default state is already declared on /datum/proc/ui_state
```suggestion
```